pandas/core/indexes/base.py

```diff
@@ -3412,7 +3412,7 @@ def _get_string_slice(self, key, use_lhs=True, use_rhs=True):
 
     def slice_indexer(self, start=None, end=None, step=None, kind=None):
         """
-        For an ordered Index, compute the slice indexer for input labels and
+        For an ordered index, compute the slice indexer for input labels and
```

Comment:
I am not fully sure it *has* to be ordered? When there are no duplicates, I think it doesn't need to be sorted:

```
In [9]: pd.Index([1, 3, 4, 2, 5]).slice_indexer(1, 2)
Out[9]: slice(0, 4, None)
```

Reply:
Maybe you can add a "Raises" section to specify when a KeyError is raised:

```
In [10]: pd.Index([1, 3, 4, 2, 5, 2]).slice_indexer(1, 2)
...
KeyError: 'Cannot get right slice bound for non-unique label: 2'

In [11]: pd.Index([1, 3, 4, 2, 5, 2]).slice_indexer(1, 4)
Out[11]: slice(0, 3, None)
```